crates/re_log_types/src/data_table.rs

```diff
@@ -854,50 +891,66 @@ impl DataTable {
     }
 
     /// Deserializes a sparse data column.
+    ///
+    /// If `compute_cell_sizes` is enabled, this will estimate the size of the arrow data within
+    /// each cell and cache it there.
+    /// See [`DataCell::compute_size_bytes`] for more information.
     fn deserialize_data_column(
         component: ComponentName,
         column: &dyn Array,
+        compute_cell_sizes: bool,
     ) -> DataTableResult<DataCellColumn> {
+        crate::profile_function!();
         Ok(DataCellColumn(
             column
                 .as_any()
                 .downcast_ref::<ListArray<i32>>()
                 .ok_or(DataTableError::NotAColumn(component.to_string()))?
                 .iter()
-                .map(|array| array.map(|values| DataCell::from_arrow(component, values)))
+                .map(|array| {
+                    array.map(|values| {
+                        let mut cell = DataCell::from_arrow(component, values);
+                        if compute_cell_sizes {
+                            cell.compute_size_bytes(); // WARN: costly!
+                        }
+                        cell
+                    })
+                })
                 .collect(),
         ))
     }
 }
 
 // ---
 
-impl TryFrom<&ArrowMsg> for DataTable {
-    type Error = DataTableError;
-
+impl DataTable {
+    /// Deserializes the contents of an [`ArrowMsg`] into a `DataTable`.
+    ///
+    /// If `compute_cell_sizes` is enabled, this will estimate the size of the arrow data within
+    /// each cell and cache it there.
+    /// See [`DataCell::compute_size_bytes`] for more information.
     #[inline]
-    fn try_from(msg: &ArrowMsg) -> DataTableResult<Self> {
+    pub fn from_arrow_msg(msg: &ArrowMsg, compute_cell_sizes: bool) -> DataTableResult<Self> {
         let ArrowMsg {
             table_id,
             timepoint_max: _,
             schema,
             chunk,
         } = msg;
 
-        Self::deserialize(*table_id, schema, chunk)
+        Self::deserialize(*table_id, schema, chunk, compute_cell_sizes)
     }
-}
-
-impl TryFrom<&DataTable> for ArrowMsg {
-    type Error = DataTableError;
 
+    /// Serializes the contents of a `DataTable` into an [`ArrowMsg`].
+    //
+    // TODO(#1760): support serializing the cell size itself, so it can be computed on the clients.
     #[inline]
-    fn try_from(table: &DataTable) -> DataTableResult<Self> {
-        let timepoint_max = table.timepoint_max();
-        let (schema, chunk) = table.serialize()?;
+    pub fn as_arrow_msg(&self) -> DataTableResult<ArrowMsg> {
```

Comment:
```suggestion
    pub fn to_arrow_msg(&self) -> DataTableResult<ArrowMsg> {
```

See https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv